solution_collector/src/main.rs

```diff
@@ -12,85 +14,135 @@ use github_rs::client::{Github, Executor};
 use serde_json::Value;
 use regex::Regex;
 use std::collections::HashMap;
+use structopt::StructOpt;
 
-#[derive(Debug)]
-enum Mode {
+#[derive(StructOpt, Debug)]
+#[structopt(name = "solution_collector")]
+struct Opt {
+    #[structopt(subcommand)]
+    command: Command
+}
+
+#[derive(StructOpt, Debug)]
+enum Command {
+    #[structopt(name = "preview")]
     Preview,
-    Merge
+    #[structopt(name = "week-report")]
+    WeekReport {
+        week: u32
+    },
+    #[structopt(name = "merge")]
+    Merge {
+        week: u32,
+    }
 }
 
-fn get_week(title: &str) -> u64 {
+struct Context {
+    owner: String,
+    repo: String,
+    client: Github
+}
+
+impl Context {
+    fn new(token: String, owner: String, repo: String) -> Self {
+        let client = Github::new(token).expect("Invalid token");
+        Context {
+            owner,
+            repo,
+            client
+        }
+    }
+}
+
+#[derive(Debug)]
+struct Problem {
```

Comment:
We need a better name so it should not be confusing. For example, `Submission` (instead of `SubmitEntry`), and `Entry` (instead of `Problem`).

Using `Problem` here makes me think about a data structure to store parser's error results.